docs/reference/transform/transforms-at-scale.asciidoc

```diff
@@ -0,0 +1,197 @@
+[role="xpack"]
+[[transform-scale]]
+= Working with {transforms} at scale
+++++
+<titleabbrev>{transforms-cap} at scale</titleabbrev>
+++++
+
+{transforms-cap} convert existing {es} indices into summarized indices, which 
+provide opportunities for new insights and analytics. The search and index 
+operations performed by {transforms} use standard Elasticsearch features so 
+similar considerations for working with Elasticsearch at scale are often 
+applicable to transforms. If you experience performance issues, start by 
+identifying the bottleneck areas (search, indexing, processing, or storage) then 
+review the relevant considerations in this guide to improve performance. It also 
+helps to understand how {transforms} work as different considerations apply 
+depending on whether or not your transform is running in continuous mode or in 
+batch.
+
+In this guide, you’ll learn how to:
+
+* Understand the impact of configuration options on the performance of 
+  {transforms}.
+
+**Prerequisites:**
+
+These guildelines assume you’re already familiar with: 
+
+* <<transform-overview,How to create {transforms}>>.
+* <<transform-setup,How to set up {transforms}>>.
+* <<transform-checkpoints,How {transform} checkpoints work in continuous mode>>.
+
+The following considerations are not sequential – the numbers help to navigate 
+between the list items; you can take action on one or more of them in any order. 
+Most of the recommendations apply to both continuous and batch {transforms}. If 
+a list item only applies to one {transform} type, this exception is highlighted 
+in the description.
+
+The keywords in parenthesis at the end of each recommendation title indicates 
+the factor that may be improved by following the given recommendation.
+
+[discrete]
+[[measure-performance]]
+== Measure {transforms} performance
+
+In order to optimize {transform} performance, start by identifying the areas 
+where most work is being done. The **Stats** interface of the 
+**{transforms-cap}** page in {kib} contains information that covers three main 
+areas: indexing, searching, and processing time (alternatively, you can use the 
+<<get-transform-stats, {transforms} stats API>>). If, for example, the results 
+show that the highest proportion of  time is spent on search, then prioritize 
+efforts on optimizing the search query of the {transform}. {transforms-cap} also 
+has https://esrally.readthedocs.io[Rally support] that makes it possible to run 
+performance checks on {transforms} configurations if it is required. If you 
+optimized the crucial factors and you still experience performance issues, you 
+may also want to consider improving your hardware.
+
+
+[discrete]
+[[frequency]]
+== 1. Optimize `frequency` (index)
```

Comment:
> == 1. Optimize `frequency` (index)

Have you considered putting all the index-related actions together? I found myself wondering why the sections flip from index-related to search-related then back to index-related suggestions. If there's a reason they're sorted that way, it wasn't obvious to me.

Reply:
The list goes from the easy wins to the steps that are harder to execute. I added a sentence to explain it (line 34-35).

Reply:
> The list goes from the easy wins to the steps that are harder to execute.  I added a sentence to explain it (line 34-35).

This isn't true enough to state .. for example limiting the scope of the query is at the bottom, but relatively easy to execute. We haven't ordered this particularly actively.

Reply:
Then I group the actions in my next commit into four groups: index, search, process, storage.

Reply:
Addressed via d7d67fa.